### PR TITLE
Revert "VectorBuilder avoids copying data if possible"

### PR DESCRIPTION
**Pull request by szeiger**
This reverts commit 18224fde06d9d5a747bdfe5603d4efe21732070d.

Reverts https://github.com/scala/scala/pull/7588 which caused
https://github.com/scala/bug/issues/11636.

See https://github.com/scala/scala/pull/8246 for context.

### CODE REVIEW

**Comment by joshlemer** [comment]
Should we add to `VectorTest.scala`?

```scala
  @Test
  def i11636(): Unit = {

    val a: Vector[String] = "O" +: Iterator.continually("E").take(2101).foldLeft(Vector.empty[String])((v, e) => v :+ e) :+ "C"
    val a0: ArraySeq[String] = ArraySeq("O") ++ Iterator.continually("E").take(2101) ++ ArraySeq("C")

    val b: Vector[String] = "O" +: Iterator.continually("E").take(223) .foldLeft(Vector.empty[String])((v, e) => v :+ e) :+ "C"
    val b0: ArraySeq[String] = ArraySeq("O") ++ Iterator.continually("E").take(223) ++ ArraySeq("C")

    val c: Vector[String] = "O" +: Iterator.continually("E").take(135) .foldLeft(Vector.empty[String])((v, e) => v :+ e) :+ "C"
    val c0: ArraySeq[String] = ArraySeq("O") ++ Iterator.continually("E").take(135) ++ ArraySeq("C")

    val d: Vector[String] = "O" +: Iterator.continually("E").take(0)   .foldLeft(Vector.empty[String])((v, e) => v :+ e) :+ "C"
    val d0: ArraySeq[String] = ArraySeq("O", "C")

    val e: Vector[String] = "O" +: Iterator.continually("E").take(376) .foldLeft(Vector.empty[String])((v, e) => v :+ e) :+ "C"
    val e0: ArraySeq[String] = ArraySeq("O") ++ Iterator.continually("E").take(376) ++ ArraySeq("C")

    val f: Vector[String] = "O" +: Iterator.continually("E").take(365) .foldLeft(Vector.empty[String])((v, e) => v :+ e) :+ "C"
    val f0: ArraySeq[String] = ArraySeq("O") ++ Iterator.continually("E").take(365) ++ ArraySeq("C")
    
    assertEquals(a0, a)
    assertEquals(b0, b)
    assertEquals(c0, c)
    assertEquals(d0, d)
    assertEquals(e0, e)
    assertEquals(f0, f)

    assertEquals(a0 ++ b0, a ++ b)
    assertEquals(a0 ++ b0 ++ c0, a ++ b ++ c)
    assertEquals(a0 ++ b0 ++ c0 ++ d0, a ++ b ++ c ++ d)
    assertEquals(a0 ++ b0 ++ c0 ++ d0 ++ e0, a ++ b ++ c ++ d ++ e)
    assertEquals(a0 ++ b0 ++ c0 ++ d0 ++ e0 ++ f0, a ++ b ++ c ++ d ++ e ++ f)
  }
```

File: src/library/scala/collection/immutable/Vector.scala [hunk 1]
```diff
@@ -110,7 +110,7 @@ object Vector extends StrictOptimizedSeqFactory[Vector] {
  *  @define mayNotTerminateInf
  *  @define willNotTerminateInf
  */
-final class Vector[+A] private[immutable] (private[collection] val startIndex: Int, private[collection] val endIndex: Int, private[immutable] val focus: Int)
```
Comment (joshlemer) [inline src/library/scala/collection/immutable/Vector.scala]: this line doesn't need to change.

**Comment by szeiger** [comment]
*sigh* the MiMa whitelist is really, really annoying. Almost every PR requires new entries which will cause conflicts with every other PR.

**Comment by joshlemer** [comment]
Thank you @szeiger

**Pull request merged**